azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py

```diff
@@ -0,0 +1,127 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azext_devops.dev.common.services import (get_new_task_agent_client,
+                                              resolve_instance,
+                                              resolve_instance_and_project)
+
+logger = get_logger(__name__)
+
+
+def list_pools(pool_name=None, pool_type=None, action=None, properties=None, organization=None, detect=None):
+    """ (PREVIEW) List agent pools
+    :param pool_name: Filter the list with matching pool name.
+    :type pool_name: str
+    :param pool_type: Filter the list with type of pool.
+    :type pool_type: str
+    :param action: Filter the list with user action permitted.
+    :type action: str
+    :param properties: Filter the list by agent pool properties. Comma separated list.
```

Comment:
An example of properties list would be helpful here

Reply:
Will ask the team owning the API about how the properties look like.. I haven't seen an example yet where the properties are returned.